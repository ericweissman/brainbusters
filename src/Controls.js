import React, { Component } from 'react';

class Controls extends Component {
  render() {
    let {studyList, showAllQuestions, toggleInstructionsShown, resetQuiz, toggle } = this.props
    if(showAllQuestions && studyList.length === 0) {
      return(
        <div className="user-controls">
          <button className="toggle-instructions-btn" onClick={toggleInstructionsShown}>Show Instructions</button>
          <button onClick={resetQuiz}>Reset Study List</button>
          <button className="cards-to-show" onClick={toggle}>Study List Empty</button>
        </div>
      )
    } else if (showAllQuestions) {
      return (
        <div className="user-controls">
          <button className="toggle-instructions-btn" onClick={toggleInstructionsShown}>Show Instructions</button>
          <button onClick={resetQuiz}>Reset Study List</button>
          <button className="cards-to-show" onClick={toggle}>Show Study List</button>
        </div>
      )
    } else if (!showAllQuestions) {
      return (
        <div className="user-controls">
          <button className="toggle-instructions-btn" onClick={toggleInstructionsShown}>Show Instructions</button>
          <button onClick={resetQuiz}>Reset Study List</button>
          <button className="cards-to-show" onClick={toggle}>Show All Questions</button>
        </div>
      )
    }

  }
}

export default Controls