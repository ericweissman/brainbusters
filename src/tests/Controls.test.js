import React from 'react';
import Controls from '../Controls';
import setupTests from '../setupTests'
import { shallow } from 'enzyme';

describe('Controls', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Controls />
    )
  })

  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });


})