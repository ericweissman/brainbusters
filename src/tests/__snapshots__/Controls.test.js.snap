// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match snapshot when all data is passed correctly 1`] = `
<div
  className="user-controls"
>
  <button
    className="toggle-instructions-btn"
  >
    Show Instructions
  </button>
  <button>
    Reset Study List
  </button>
  <button
    className="cards-to-show"
  >
    Show All Questions
  </button>
</div>
`;
